static/src/javascripts/projects/common/modules/commercial/adblock-ask.js

```diff
@@ -25,35 +29,19 @@ const askHtml = `
 </div>
 `;
 
-export const adblockTest: ABTest = {
-    id: 'AdblockAsk',
-    start: '2019-02-20',
-    expiry: '2021-02-16',
-    author: 'Tom Forbes',
-    description:
-        'Places a contributions ask underneath the right-hand ad slot on articles.',
-    audience: 1,
-    audienceOffset: 0,
-    successMeasure: '',
-    audienceCriteria: '',
-    showForSensitive: true,
-    canRun() {
-        return (
-            !shouldHideSupportMessaging() &&
-            !pageShouldHideReaderRevenue() &&
-            !config.get('page.hasShowcaseMainElement')
-        );
-    },
+const canShow = () =>
+    !shouldHideSupportMessaging() &&
+    !pageShouldHideReaderRevenue() &&
+    !config.get('page.hasShowcaseMainElement');
 
-    variants: [
-        {
-            id: 'control',
-            test: (): void => {
-                const slot = document.querySelector('.js-aside-slot-container');
-                if (slot) {
-                    slot.innerHTML += askHtml;
-                }
-            },
-        },
-    ],
+export const initAdblockAsk = () => {
+    fastdom
+        .read(() => $('.js-aside-slot-container'))
+        .then(slot => {
+            if (canShow()) {
+                return fastdom.write(() => {
+                    slot.append(askHtml);
```

Comment:
I am not 100% sure what the behaviour of this is if slot is null or undefined - should we add an additional check i.e. `if (slot) { ... }`

Reply:
Isn't that what `.read` is doing?

Reply:
I don't know for sure, but it might be that fastdom will return `slot = null`, rather than reject the promise (which wouldn't even enter the `.then`). If this is the case then `(null).append` will be called and throw an exception.

Reply:
Okay, fair enough 😄